### ISSUE TITLE
src/editor/libs/codemirror/ is empty in downloadable source release

### ISSUE THREAD

**Issue by ericjs** [notepadqq/notepadqq#452]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/notepadqq/notepadqq/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Building from source downloaded from https://github.com/notepadqq/notepadqq/archive/v1.0.1.tar.gz
fails because src/editor/libs/codemirror/ is empty.

**Expected behavior**
src/editor/libs/codemirror/ is not empty and the build does not fail

**Minimal reproduction of the problem with instructions**
Download from link above, extract, cd to the right directory and run
./configure --prefix /usr --qmake /usr/bin/qmake-qt5 --lrelease /usr/bin/lrelease-qt5
(successful)
the command fails with 
error: CodeMirror submodule not available at `src/editor/libs/codemirror'! Make sure you downloaded the source code using git as specified in README.md

Note there are no such instructions in the README.md. And surely this is not intended anyway with this downloadable source package?

**Please tell us about your environment:**
Chakra Linux, QT 5.8, Notepadqq 1.0.1

**Comment by ericjs** [comment]
Also your Makefile-codemirror tries to
cd "$(DESTDIR)"/mode/m4/

I tried manually adding a few recent codemirror releases and they have no m4 directory in their mode directory, so this fails.

**Comment by ericjs** [comment]
It case it wasn't clear, this issue was about the latest release src from https://github.com/notepadqq/notepadqq/releases not being buildable without manual intervention (and taking  some guesses). I made it work on my local machine, and would have submitted it to my distro's communuty repository, but not in that state (i.e. with my own hacks, without even any comment from the npqq team as to the correct version of codemirror to use).

**Comment by JuBan1** [comment]
Sorry for not getting a reply for so long. The project is on a low burn for the time being.

Deployment isn't exactly my specialty but I'll try to clear things up as best I can:

> Note there are no such instructions in the README.md. And surely this is not intended anyway with this downloadable source package?

The instructions are to download the source from git. The README provides the following commands:

> $ git clone --recursive https://github.com/notepadqq/notepadqq.git
    $ cd notepadqq
    notepadqq$ ./configure --prefix /usr
    notepadqq$ make

The `--recursive` git-clone also initializes the CodeMirror submodule. That also creates the `$(DESTDIR)"/mode/m4/` directory. Following these instructions you should have no problem getting the full source or installing Nqq.

You're right that the source package is messed up because `configure` depends on the Nqq git repo being there. I'll brush up on my bash and see if I can't modify `configure` to grab the submodule regardless of the repo's presence.

**Comment by JuBan1** [comment]
Fixed in #c35921d and following.
